packages/extension-ui/src/Popup/Signing/Extrinsic.tsx

```diff
@@ -92,52 +53,158 @@ function Extrinsic({
 
 const FullWidthTable = styled(Table)`
   width: 100%;
+  
+  ${EllipsisCell} { /* Styles declared as a descendant of the <Table> in order to override the intrusive <Table> styles */
+    display: block;
+    text-overflow: ellipsis;
+    overflow: hidden;
+    white-space: nowrap;
+  }
+`;
+
+const HeaderCell = styled.td`
+  white-space: nowrap;
+`;
+
+const BatchRow = styled.tr`
+  margin-top: 40px;
 `;
+
+const CallDefinitionRows = (props: { section: string, method: string, args: { name: string, value?: Renderable }[], docs?: string[]}) => {
+  const { t } = useTranslation();
+
+  return (
+    <>
+      <tr>
+        <td className='label'>{t<string>('Section')}</td>
+        <div className='separator'></div>
+        <EllipsisCell className='data'>{props.section}</EllipsisCell>
+      </tr>
+      <tr>
+        <td className='label'>{t<string>('Method')}</td>
+        <div className='separator'></div>
+        <EllipsisCell className='data'>{props.method}</EllipsisCell>
+      </tr>
+      {props.args?.length && (
+        <tr>
+          <div className='separator'></div>
+          <HeaderCell className='label'>{t<string>('Arguments')}</HeaderCell>
+          <div className='separator'></div>
+        </tr>
+      )}
+      {props.args.map((arg) => (
+        <tr key={arg.name}>
+          <td className='label'>{arg.name}</td>
+          <div className='separator'></div>
+          <EllipsisCell className='data'>{arg.value ?? '-'}</EllipsisCell>
+        </tr>
+      ))}
+    </>
+  );
+};
+
+/**
+ * Returns call details for a singular call and an array of call details for a batch.
+ */
+const formatCall = (call: Call) => {
+  const isBatch = call.args[0]?.toRawType().startsWith('Vec<Call>');
+
+  if (isBatch) {
+    return (call.args[0] as unknown as Call[]).map((callFromBatch: Call) => ({
+      section: callFromBatch.section,
+      method: callFromBatch.method,
+      args: formatCallArgs(callFromBatch),
+      docs: getCallDocs(callFromBatch),
+    }));
+  }
+
+  return {
+    section: call.section,
+    method: call.method,
+    args: formatCallArgs(call),
+    docs: getCallDocs(call),
+  };
+};
+
+const formatCallArgs = (call: Call) =>
+  call.meta.args.flatMap((argMeta, i) => {
+    const argName = argMeta.name.toHuman();
+
+    try {
+      const value = call.args[i];
+
+      // We're not displaying a transaction parameter without an argument passed to the transaction
+      if (value.isEmpty) {
+        return [];
+      }
+
+      const argType = argMeta.typeName.toString();
+      const formatter = formatters[argType] || defaultFormatter;
+
+      return [{ name: argName, value: formatter(value) }];
+    } catch (e) {
+      console.error(`Failed argument "${argName}" decoding:`, e);
+
+      // In the case of an unlikely argument decoding error, for clarity we still want to display that this argument exists
+      return [{ name: argName }];
+    }
+  });
+
+type Renderable = string | number;
```

Comment:
I thought it was quite descriptive :D As in _anything that you can render, cause it's going to be rendered_, so in case anyone wants to add anything to this union in the future, he know it can't be sth like an object or symbol etc.

Reply:
I don't mind `Renderable` if you like it, but I wouldn't expect it to be `string | number`

Reply:
I don't mind `Renderable`, but I'd probably keep it as `string | number` :sweat_smile: Didn't mean make an issue out of this, whatever you prefer

Reply:
Ok, I'll leave it as is then :)